src/components/Interests.js

```diff
@@ -1,32 +1,117 @@
+import Checkbox from "react-native-modest-checkbox";
+
+const jsonData = {
+  "items": [
+     {"name" : "Electronics"} ,
+     {"name" : "Food"},
+     {"name" : "Clothes"}
+  ]
+};
 
 export default class Interests extends Component{
      constructor(props){
             super(props);
             console.log(this.props);
 
             this.state = {
-              check1 : false,
-              check2 : false
+              options : [],
+              //checked : true
+              userid : ""
             }
-
             }
 
+    getOptions = async ()=> {
+       try{
+           //let response = await fetch('http://192.168.43.102:8083/partner/interests/')
+           //let res = await response.json();
+           //if (response.status >= 200 && response.status < 300){
+               console.log("am in if oondition");
+               this.setState({
+                 options: jsonData.items.map((item) => ({
+                     title : item.name,
+                     checked : true
+                     })
+                     )}
+                     );
+               //} else {
+                // let error = res;
+                 //throw error;
+               } catch(error) {
+                console.log("error from server --> ", error);
+                alert("Failed to connect to server");
+
+               }
+
+      }
+
+
+   async componentDidMount(){
+       this.getOptions();
+       try {
+               const value = await AsyncStorage.getItem('user_id');
+               if (value !== null) {
+                  // We have data!!
+                  console.log(" inside interests component did mount ", value)
+                  this.setState({ userid : Number(value) });
+                  } else {
+                    console.log('No value returned from storage');
+                    alert("Userid cannot be found")
+                  }
+
+                      } catch (error) {
+                        console.log("unable to fetch the value")
+                        // Handle errors here
+                      }
+    }
+
+    selectInterest(checked, index) {
+        const { options } = this.state;
+        options[index].checked = !options[index].checked;
+        this.setState({ options: options });
+        console.log("options ", this.state.options);
+      }
+
+
+     updateInterests = async () => {
+        let interests = this.state.options.filter(item =>item.checked).map((item=>item.title));
+        console.log(interests);
+        let url = "http://192.168.43.102:8083/user/interest/"
+        await fetch(`${url}${this.state.userid}`,{
+                         method: 'PATCH',
+                         headers: {
+                            'Accept': 'application/json',
+                            'Content-Type': 'application/json',
+                            },
+                         body: JSON.stringify({
+                           "user_interests" : interests
```

Comment:
There is a change pushed in user service for user_interests. It now expects a ArrayList<String>. Please test again and change accordingly.

Reply:
Updated. passes interests as array.